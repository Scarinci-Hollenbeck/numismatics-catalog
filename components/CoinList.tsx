import React from 'react';
import BreadCrumbs from './BreadCrumbs';
import CoinListItem from './CoinListItem';
import { LinkItem, BreadCrumb } from '../interfaces';

type Props = {
  coinList: LinkItem[]
  collectionName: string
  breadCrumbLinks: BreadCrumb[]
}
export default function CoinList({
  collectionName,
  coinList,
  breadCrumbLinks,
}: Props): JSX.Element {
  return (
    <>
      <BreadCrumbs crumbs={breadCrumbLinks} />
      <section className="coins">
        <h2>
          List of
          {collectionName}
          {' '}
          coins
        </h2>
        <hr />
        <ul className="list">
          {coinList.map((coin) => (
            <CoinListItem key={coin._id} coin={coin} authed={false} />
          ))}
        </ul>
        <style jsx>
          {`
            section {
              background-color: #fff;
              box-shadow: 2px 4px 20px #a9a9a9;
              max-width: 100%;
              padding: 1em;
              margin-top: 3em;
            }

            section h2 {
              font-family: 'Tajawal Regular';
              font-size: 2rem;
              margin: 0;
              padding: 0;
              text-align: left;
              color: #2980b9;
            }

            section hr {
              background-color: #2980b9;
              border: 0;
              height: 5px;
              border-radius: 5px;
            }

            section ul {
              margin: 0;
              padding: 0;
              text-align: left;
              margin-top: 1em;
            }
          `}
        </style>
      </section>
    </>
  );
}
