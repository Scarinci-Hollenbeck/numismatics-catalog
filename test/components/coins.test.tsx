import React from 'react';
import renderer from 'react-test-renderer';
import CoinLisItem from '../../components/CoinListItem';
import CoinList from '../../components/CoinList';

const collectionName = 'Calico Kings and Queens of Spain';

const breadCrumbLinks = [
  'Calico Kings and Queens of Spain',
  'King Ferdinand VI, 1746 - 1759',
];

const coinList = [
  {
    id: '1234567',
    title: 'King Ferdinand VI, 1746 - 1759',
    image: 'https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg',
    description:
      'Obv: Ferdin. VI D.G. Hisp. Et Ind. R. 1746-1759, Rev: Aeqvilibrivm Weight: 63.7864 Grams, 2.25 Ounces Measurements: .50 MM Diameter, 2 Inches In Diameter',
    link: 'king-ferdinand-vi-1746-1759',
  },
  {
    id: '1234568',
    title: 'King Ferdinand VI, 1746 - 1759',
    image: 'https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg',
    description:
      'Obv: Ferdin. VI D.G. Hisp. Et Ind. R. 1746-1759, Rev: Aeqvilibrivm Weight: 63.7864 Grams, 2.25 Ounces Measurements: .50 MM Diameter, 2 Inches In Diameter',
    link: 'king-ferdinand-vi-1746-1759',
  },
  {
    id: '1234569',
    title: 'King Ferdinand VI, 1746 - 1759',
    image: 'https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg',
    description:
      'Obv: Ferdin. VI D.G. Hisp. Et Ind. R. 1746-1759, Rev: Aeqvilibrivm Weight: 63.7864 Grams, 2.25 Ounces Measurements: .50 MM Diameter, 2 Inches In Diameter',
    link: 'king-ferdinand-vi-1746-1759',
  },
];

describe('Coin list Item component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<CoinLisItem coin={coinList[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Coin list Container component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <CoinList
          collectionName={collectionName}
          coinList={coinList}
          breadCrumbLinks={breadCrumbLinks}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
