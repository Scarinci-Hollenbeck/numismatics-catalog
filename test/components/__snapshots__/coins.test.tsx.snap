// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin list Container component matches snapshot 1`] = `
Array [
  <ul
    className="breadcrumbs"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/Calico%20Kings%20and%20Queens%20of%20Spain"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Calico Kings and Queens of Spain
      </a>
    </li>
    <li>
      <a
        href="/King%20Ferdinand%20VI,%201746%20-%201759"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        King Ferdinand VI, 1746   1759
      </a>
    </li>
  </ul>,
  <section
    className="coins"
  >
    <h2>
      List of 
      Calico Kings and Queens of Spain
       coins
    </h2>
    <hr />
    <ul
      className="list"
    >
      <li
        className="coin-item"
      >
        <a
          href="/coin/1234567/king-ferdinand-vi-1746-1759"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          King Ferdinand VI, 1746 - 1759
        </a>
      </li>
      <li
        className="coin-item"
      >
        <a
          href="/coin/1234568/king-ferdinand-vi-1746-1759"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          King Ferdinand VI, 1746 - 1759
        </a>
      </li>
      <li
        className="coin-item"
      >
        <a
          href="/coin/1234569/king-ferdinand-vi-1746-1759"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          King Ferdinand VI, 1746 - 1759
        </a>
      </li>
    </ul>
  </section>,
]
`;

exports[`Coin list Item component matches snapshot 1`] = `
<li
  className="coin-item"
>
  <a
    href="/coin/1234567/king-ferdinand-vi-1746-1759"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    King Ferdinand VI, 1746 - 1759
  </a>
</li>
`;
